packages/openapi-react-query/src/index.ts

```diff
@@ -128,12 +171,49 @@ export default function createClient<Paths extends {}, Media extends MediaType =
     ...options,
   });
 
+  const infiniteQueryOptions = <
+    Method extends HttpMethod,
+    Path extends PathsWithMethod<Paths, Method>,
+    Init extends MaybeOptionalInit<Paths[Path], Method & keyof Paths[Path]>,
+    Response extends Required<FetchResponse<any, Init, Media>>,
+  >(
+    method: Method,
+    path: Path,
+    init?: InitWithUnknowns<Init>,
+    options?: Omit<
+      UseInfiniteQueryOptions<
+        Response["data"],
+        Response["error"],
+        Response["data"],
+        number,
+        QueryKey<Paths, Method, Path>
+      >,
+      "queryKey" | "queryFn"
+    >,
+  ) => ({
+    queryKey: [method, path, init] as const,
+    queryFn,
+    ...options,
+  });
+
   return {
     queryOptions,
     useQuery: (method, path, ...[init, options, queryClient]) =>
       useQuery(queryOptions(method, path, init as InitWithUnknowns<typeof init>, options), queryClient),
     useSuspenseQuery: (method, path, ...[init, options, queryClient]) =>
       useSuspenseQuery(queryOptions(method, path, init as InitWithUnknowns<typeof init>, options), queryClient),
+    useInfiniteQuery: (method, path, ...[init, options, queryClient]) => {
+      const baseOptions = infiniteQueryOptions(method, path, init as InitWithUnknowns<typeof init>, options as any); // TODO: find a way to avoid as any
+      return useInfiniteQuery(
+        {
+          ...baseOptions,
+          initialPageParam: 0,
+          getNextPageParam: (lastPage: any, allPages: any[], lastPageParam: number, allPageParams: number[]) =>
```

Comment:
Do we need to add `getPreviousPageParam` here too or do we get it by default via `baseOptions` ?